src/main/scala/scorex/util/serialization/VLQWriter.scala

```diff
@@ -1,23 +1,43 @@
 package scorex.util.serialization
 
+import java.util
 import java.util.{Arrays, BitSet}
+
 import scorex.util.encode.ZigZagEncoder._
 
 trait VLQWriter extends Writer {
 
   def toBytes: Array[Byte]
 
+  /**
+    * Encode signed Short using ZigZag and then VLQ.
+    * Both negative and positive values are supported, but due to ZigZag encoding positive
+    * values is done less efficiently than by [[putUShort]].
+    * Use [[putUShort]] to encode values that are always positive.
+    *
+    * @see [[https://en.wikipedia.org/wiki/Variable-length_quantity]]
+    * @note Have to be decoded '''only''' with [[VLQReader.getShort]]
+    *       The resulting varint uses ZigZag encoding, which is much more efficient at
+    *       encoding negative values than pure VLQ.
+    * @param x signed Short
+    */
+  @inline override def putShort(x: Short): this.type = {
+    if (x < Short.MinValue || x > Short.MaxValue)
```

Comment:
This check is not needed, as the value in the Short domain.

Reply:
fixed